src/nspawn/nspawn.c

```diff
@@ -3824,14 +3861,23 @@ int main(int argc, char *argv[]) {
                                 goto finish;
 
                         if (arg_network_veth) {
-                                r = setup_veth(arg_machine, pid, veth_name, !!arg_network_bridge);
+                                r = setup_veth(arg_machine, pid, veth_name,
+                                               arg_network_bridge || arg_network_zone);
```

Comment:
If the veth already exists, this will fail with -EEXIST, and then the previously existing interface will be removed. One way to trigger this is to have two containers with long names, so that the veth name that is used will be identical for both. Starting the second container will fail here, but the interface of the first container will be dropped too.

I guess that this issue already existed before, but it would be nice to fix this. Maybe it should be another PR.